onnxruntime/core/graph/graph_utils.h

```diff
@@ -86,20 +81,49 @@ bool GetRepeatedNodeAttributeValues(const Node& node,
   return false;
 }
 
-/** Removes the given Node from the Graph and keeps Graph consistent by rebuilding needed connections.
+/** Check if it will be possible to remove or fuse a node.
     We support the removal of the Node as long as the following conditions hold:
-    - There should be no implicit inputs.
-    - Only one of the outputs is used by downstream operators (but it can have multiple output edges).
-    - If the Node has a single incoming node (and possibly multiple initializers), we can remove the Node and
-      connect its incoming node to its outgoing nodes.
-    - If the Node has a single initializer as input, we remove the Node and feed the initializer as input to its
-      output nodes. */
-bool RemoveNode(Graph& graph, Node& node);
+    - The node should not produce a graph output unless something else will produce that output.
+    - Only one of the outputs is used by downstream operators (multiple output edges are allowed).
+    - If the Node has a single incoming node, we can remove the Node and connect its incoming node to its 
+      outgoing nodes, if doing so does not clash with any values in any relevant subgraphs. 
```

Comment:
Does this mean that it can have other input node args that are not nodes?
Instead of "we can remove that node", we can say something like "this condition is required so that we can connect the incoming node to the outgoing nodes after removing the node". Or we can move this comment to the RemoveNodeAndUpdateEdges.
The clash with the subgraphs happens when the output of the node to be removed is used by a subgraph? Let's mention it.

Reply:
The existing check does allow multiple input nodeargs when there is one edge, but not sure if that was intentional and should be allowed or not. Given that check is generally used when we're wiring together the input and output nodes, having multiple inputs seems invalid.

Maybe CanRemoveNode is trying to do too many things and we should separate node removal from replacement.